### PR TITLE
Refactor architecture backend - services, models, db, manager, unit testing

### PR DESCRIPTION
**Pull request by alfredovargasjr**
# Description

This PR includes the refactoring of the project architecture for the backend. It also includes additions and modifications to services, managers, and db models.

## Type of change

- model changes
- services
- managers
- migrations

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] UT: password services
# Notes

Db is a local mssqlserver, run a instance of the db context to create it.

# Pivotal Tracker Link:

None

### CODE REVIEW

File: Backend/DataAccessLayer/Migrations/201812011641236_create-table-service.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        public override void Up()
+        {
+            CreateTable(
+                "dbo.Services",
+                c => new
+                {
+                    Id = c.Guid(nullable: false),
+                    ServiceName = c.String(nullable: false),
+                    Disable = c.Boolean(nullable: false),
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201812011641236_create-table-service.cs]: Needs to be "disabled"

File: Backend/DataAccessLayer/Migrations/201812011646551_create-table-session.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            CreateTable(
+                "dbo.Sessions",
+                c => new
+                {
+                    Id = c.Guid(nullable: false),
+                    Token = c.String(nullable: false),
+                    ExpiresAt = c.DateTime(nullable: false, precision: 7, storeType: "datetime2"),
+                    UpdatedAt = c.DateTime(nullable: false, precision: 7, storeType: "datetime2"),
+                    CreateAt = c.DateTime(nullable: false, precision: 7, storeType: "datetime2"),
+                    UserInSession = c.Guid(nullable: false),
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201812011646551_create-table-session.cs]: Should be "userId"

File: Backend/DataAccessLayer/Migrations/201812021657569_update-claim-AddCreatedAtAddUpdateAt.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+namespace DataAccessLayer.Migrations
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201812021657569_update-claim-AddCreatedAtAddUpdateAt.cs]: This migration should not exist. Merge this into the initial table creation

File: Backend/DataAccessLayer/Migrations/201812021702408_update-service-AddCreateatAddUpdatedatMakeDisableRequired.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+namespace DataAccessLayer.Migrations
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201812021702408_update-service-AddCreateatAddUpdatedatMakeDisableRequired.cs]: Same for this

File: Backend/DataAccessLayer/Migrations/201812021901246_update-user-AddSecQuestionsAddSecAnswersColumns.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+namespace DataAccessLayer.Migrations
+{
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201812021901246_update-user-AddSecQuestionsAddSecAnswersColumns.cs]: Same for this

File: Backend/DataAccessLayer/Migrations/201812021719031_update-user-RemoveColumnNameUserInSession.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
+namespace DataAccessLayer.Migrations
```
Comment (julianpoy) [inline Backend/DataAccessLayer/Migrations/201812021719031_update-user-RemoveColumnNameUserInSession.cs]: Same for this

**Review by julianpoy — CHANGES_REQUESTED** [review]

File: Backend/DataAccessLayer/Migrations/201812021657569_update-claim-AddCreatedAtAddUpdateAt.cs (hunk 3 above)
Comment (alfredovargasjr) [reply Backend/DataAccessLayer/Migrations/201812021657569_update-claim-AddCreatedAtAddUpdateAt.cs]: shouldnt migrations be used for any updates to tables

**Comment by julianpoy** [comment]
Merge conflicts still exist

**Review by julianpoy — APPROVED** [review]

**Comment by julianpoy** [comment]
Approval for when you fix the merge conflicts.

**Pull request merged**